v03_pipeline/lib/tasks/dataproc/create_dataproc_cluster_test.py

```diff
@@ -0,0 +1,125 @@
+import unittest
+from types import SimpleNamespace
+from unittest.mock import Mock, call, patch
+
+import google.api_core.exceptions
+import luigi
+
+from v03_pipeline.lib.model import DatasetType, ReferenceGenome
+from v03_pipeline.lib.tasks.dataproc.create_dataproc_cluster import (
+    CreateDataprocClusterTask,
+)
+
+
+@patch(
+    'v03_pipeline.lib.tasks.dataproc.create_dataproc_cluster.dataproc.ClusterControllerClient',
+)
+class CreateDataprocClusterTaskTest(unittest.TestCase):
+    def test_dataset_type_unsupported(self, mock_cluster_controller: Mock) -> None:
+        worker = luigi.worker.Worker()
+        task = CreateDataprocClusterTask(
+            reference_genome=ReferenceGenome.GRCh38,
+            dataset_type=DatasetType.MITO,
+            run_id='1',
+        )
+        worker.add(task)
+        worker.run()
+        self.assertTrue(task.complete())
+
+    def test_spinup_cluster_already_exists_failed(
+        self,
+        mock_cluster_controller: Mock,
+    ) -> None:
+        mock_client = mock_cluster_controller.return_value
+        mock_client.get_cluster.return_value = SimpleNamespace(
```

Comment:
nice I just looked up `SimpleNamespace` and it seems useful! would a dictionary work here too?

Reply:
they're attributes on the returned object from the google cloud api... so dictionary "__getitem__" syntax doesn't work.  very annoying that there isn't a clearer way to do this in python